### ISSUE TITLE
Issue with connecting to server

### ISSUE THREAD

**Issue by amrithathorath** [cisco/libacvp#283]
I'm trying to build the latest version of software and seem to have difficulties connecting to server.
I've shown the error and set up below:

$ ./app/acvp_app --all_algs --sample

Using the following parameters:

>     ACV_SERVER:     demo.acvp.nist.gov
>     ACV_PORT:       443
>     ACV_URI_PREFIX: acvp/v1/
>     ACV_CA_FILE:    certs/acvp.nist.gov.crt
>     ACV_CERT_FILE:  certs/CSR_athorath-corsec.cer
>     ACV_KEY_FILE:   certs/privateKey.pem
> 
> ***ACVP [ERR][acvp_curl_http_post:323]--> Curl failed with code 6 (Couldn't resolve host name)
> 
> ***ACVP [STATUS][log_network_status:885]--> POST Login...
>         Status: 0
>         Url: https://demo.acvp.nist.gov:443/acvp/v1/login
>         Resp:
> (null)
> 
> ***ACVP [STATUS][acvp_register:1396]--> Login Send Failed
> ***ACVP [ERR][acvp_run:2142]--> Failed to register with ACVP server

I tried to use "acvp/" for ACV_URI_PREFIX (excluded /v1/) but still can't connect.

> $ ./app/acvp_app --all_algs --sample
> 
> Using the following parameters:
> 
>     ACV_SERVER:     demo.acvp.nist.gov
>     ACV_PORT:       443
>     ACV_URI_PREFIX: acvp/
>     ACV_CA_FILE:    certs/acvp.nist.gov.crt
>     ACV_CERT_FILE:  certs/CSR_athorath-corsec.cer
>     ACV_KEY_FILE:   certs/privateKey.pem
> 
> ***ACVP [ERR][acvp_curl_http_post:323]--> Curl failed with code 6 (Couldn't resolve host name)
> 
> ***ACVP [STATUS][log_network_status:885]--> POST Login...
>         Status: 0
>         Url: https://demo.acvp.nist.gov:443/acvp/login
>         Resp:
> (null)
> 
> ***ACVP [STATUS][acvp_register:1396]--> Login Send Failed
> ***ACVP [ERR][acvp_run:2142]--> Failed to register with ACVP server

I've run the client before and there's no change in my keys or certs.

Curl version and LD_LIBRARY_PATH shown below:

> $ echo $LD_LIBRARY_PATH
> /home/amritha/cavp_tool/curl-7.59.0/lib:/home/amritha/ssl/lib
> 
> $ curl -V
> curl 7.59.0 (x86_64-pc-linux-gnu) libcurl/7.59.0 OpenSSL/1.0.2p zlib/1.2.8 librtmp/2.3
> Release-Date: 2018-03-14
> Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp smb smbs smtp smtps telnet tftp
> Features: AsynchDNS IPv6 Largefile NTLM NTLM_WB SSL libz TLS-SRP UnixSockets HTTPS-proxy

**Comment by bfussell** [comment]
Curl failed with code 6 (Couldn't resolve host name)

Because your ACV_SERVER should be demo.acvts.nist.gov not demo.acvp.nist.gov

I think it is wrong in the nist_setup.sh script, I'll fix that.

**Comment by amrithathorath** [comment]
@bfussell : Thanks for the comment. I've done that and I still an issue.

amritha@thorath-vm:~/libacvp/libacvp-master$ ./app/acvp_app --all_algs --sample

Using the following parameters:

>     ACV_SERVER:     demo.acvts.nist.gov
>     ACV_PORT:       443
>     ACV_URI_PREFIX: acvp/
>     ACV_CA_FILE:    certs/acvp.nist.gov.crt
>     ACV_CERT_FILE:  certs/CSR_athorath-corsec.cer
>     ACV_KEY_FILE:   certs/privateKey.pem
> 
> ***ACVP [ERR][acvp_curl_http_post:323]--> Curl failed with code 60 (Peer certificate cannot be authenticated with given CA certificates)
> 
> ***ACVP [STATUS][log_network_status:885]--> POST Login...
>         Status: 0
>         Url: https://demo.acvts.nist.gov:443/acvp/login
>         Resp:
> (null)
> 
> ***ACVP [STATUS][acvp_register:1396]--> Login Send Failed
> ***ACVP [ERR][acvp_run:2142]--> Failed to register with ACVP server

I tried to see if I can access the URL https://demo.acvts.nist.gov:443/acvp/login on browser and it says "Forbidden" and I'm guessing this has something to do with the email Jason sent out this morning after maintenance?

**Comment by bfussell** [comment]
You definitely got passed the DNS name issue and now have a CA cert issue.
That CA cert doesn't work for me either. I always use the mozilla CA cert from the
certs/ directory

**Comment by amrithathorath** [comment]
@bfussell : Thanks for the pointer about Mozilla cert.
I think I got past that error. And now I see this:

> $ ./app/acvp_app --all_algs --sample
> 
> Using the following parameters:
> 
>     ACV_SERVER:     demo.acvts.nist.gov
>     ACV_PORT:       443
>     ACV_URI_PREFIX: acvp/
>     ACV_CA_FILE:    certs/mozzila_trust_anchors.pem
>     ACV_CERT_FILE:  certs/CSR_athorath-corsec.cer
>     ACV_KEY_FILE:   certs/privateKey.pem
> 
> ***ACVP [STATUS][log_network_status:885]--> POST Login...
>         Status: 404
>         Url: https://demo.acvts.nist.gov:443/acvp/login
>         Resp:
> <html><head><title>Error</title></head><body>Not Found</body></html>
> 
> ***ACVP [STATUS][acvp_register:1396]--> Login Send Failed
> ***ACVP [ERR][acvp_run:2142]--> Failed to register with ACVP server

I'm trying to request sample files for all algs. Is there anything wrong with the args I'm passing which caused the issue?

**Comment by bfussell** [comment]
You forgot to change ACV_URI_PREFIX back

**Comment by bfussell** [comment]
BTW, the best way to get a full set of vectors is to use the new offline mode options.

/app/acvp_app --all_algs --vector_req req.json

This way the client won't bother trying to process them.

**Comment by amrithathorath** [comment]
Oops my bad, I've corrected the ACV_URI_PREFIX.
So I've tried to use the command you gave in previous comment and I see:

> {
>   "acvVersion" : "1.0",
>   "error" : "The supplied one time password could not be verified"
> }

I know sometime back there was an issue with seed generation and null byte not being handled or something on those lines. I'm checking with NIST to see if the TOTP seed I received a few months back is still valid.
Meanwhile I wanted to check I didn't make any other mistake.
I tried to export ACV_TOTP_SEED with and without quotes.
